### PR TITLE
Messaging Queue Stats

### PR DESCRIPTION
**Pull request by problemv**
For #836 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/zaqar/blob/8e0dd9763aba67bb7094cf92312daeba8d087866/zaqar/transport/wsgi/v2_0/stats.py#L38

### CODE REVIEW

**Comment by problemv** [comment]
@jtopjian This is ready for review.

File: openstack/messaging/v2/queues/doc.go [hunk 1]
```diff
@@ -75,5 +75,12 @@ Example to Delete a Queue
 	if err != nil {
 		panic(err)
 	}
+
+Example to GetStat a Queue
```
Comment (jtopjian) [inline openstack/messaging/v2/queues/doc.go]: Example to Get Message Stats of a Queue

File: openstack/messaging/v2/queues/results.go [hunk 2]
```diff
@@ -33,6 +33,11 @@ type GetResult struct {
 	commonResult
 }
 
+// StatResult contains the result of a Share operation.
+type StatResult struct {
+	commonResult
```
Comment (jtopjian) [inline openstack/messaging/v2/queues/results.go]: gophercloud.Result

File: openstack/messaging/v2/queues/results.go [hunk 3]
```diff
@@ -77,13 +82,34 @@ type QueueDetails struct {
...
 func (r commonResult) Extract() (QueueDetails, error) {
 	var s QueueDetails
 	err := r.ExtractInto(&s)
 	return s, err
 }
 
+// Extract interprets any StatResult as a Stats.
+func (r StatResult) Extract() (Stats, error) {
+	var s struct {
+		Messages Stats `json:"messages"`
```
Comment (jtopjian) [inline openstack/messaging/v2/queues/results.go]:
```go
Stats Stats `json:"messages"`
```

`Messages` could infer actual messages would be returned. I understand `messages` is the JSON tag, though.

**Comment by jtopjian** [comment]
@problemv Also, since the prior commits have been merged, would you mind doing a rebase and then a force push with the single commit (changes requested can be rolled up into the single commit).

**Comment by problemv** [comment]
Sure thing.

**Comment by problemv** [comment]
@jtopjian Rebased and fixed.

**Comment by jtopjian** [comment]
LGTM - thank you :)

**Pull request merged**